### PR TITLE
Update telemetry/metrics to support adding histogram view

### PR DESCRIPTION
**Pull request by yfei1**
This commit updated `telemetry/metrics.go` file to support Prometheus's histogram view. Also added a histogram metric to monitor Redis connection latency.

Will rename`IncrementCounterN` and `IncrementCounter` to a more general name in a later PR as it is applicable to `gauge/counter/histogram` views

### CODE REVIEW

File: internal/telemetry/metrics.go [hunk 1]
```diff
@@ -45,19 +45,42 @@ func Counter(name string, description string, tags ...tag.Key) *stats.Int64Measu
...
+	histogramView(s, bounds, tags...)
+	return s
+}
+
 // IncrementCounter +1's the counter metric.
 func IncrementCounter(ctx context.Context, s *stats.Int64Measure, tags ...tag.Mutator) {
 	IncrementCounterN(ctx, s, 1, tags...)
 }
 
 // IncrementCounterN increases a metric by n.
```
Comment (Laremere) [inline internal/telemetry/metrics.go]: I don't understand this comment, in its relationship to the histogram.  I would expect, eg, that if I handled 2 tickets, I would increment the tickets handled by 2 using this command.  I'm not seeing how this related to histogram usage.  When I made a redis connection, why am I incrementing this by the time it took, eg, if it took 100ms, I should be updating the relevant bucket as +1, not adding 100 times connected to the total connected count.

Comment (yfei1) [reply internal/telemetry/metrics.go]: See the commit description for explanation. I'll do the rename in a later PR. Actually `IncrementCounter` is not an appropriate name for this function. What `RecordWithTags` does is `records one or multiple measurements at once`.

Comment (Laremere) [reply internal/telemetry/metrics.go]: Generally, I would prefer the rename be done first.  Keep that in mind in the future.  For now, update the comment to be accurate, at least.

Comment (yfei1) [reply internal/telemetry/metrics.go]: Done. Will do the rename in the next PR.

**Pull request merged**